TicTacToe/TicTacToe.cs

```diff
@@ -14,64 +15,120 @@ class Program
 
         public static void Main()
         {
-            do
+
+            while (!CheckForWin() && !CheckForTie())
             {
                 DrawBoard();
                 GetInput();
 
-            } while (!CheckForWin() && !CheckForTie());
+            }
 
             // leave this command at the end so your program does not close automatically
             Console.ReadLine();
         }
 
         public static void GetInput()
-        {
+        {   
+            
             Console.WriteLine("Player " + playerTurn);
             Console.WriteLine("Enter Row:");
             int row = int.Parse(Console.ReadLine());
             Console.WriteLine("Enter Column:");
             int column = int.Parse(Console.ReadLine());
+            PlaceMark(row, column);//pass row and column to PlaceMark method
         }
 
         public static void PlaceMark(int row, int column)
         {
-        // your code goes here
+
+            // when the user enters row and column on the already filled cell, the entry will not be accepted.
+            if (board[row][column] == "X" || board[row][column] == "O")
+            {
+                Console.WriteLine("This cell is already filled. Try again");
+            }
+            else
+            {//when the user enters a mark on an empty cell, it will be marked and player will take turn.
+             //before player turns, check if a player wins.
+
+                board[row][column] = playerTurn;
+
+                playerTurn = (playerTurn == "X") ? "O" : "X";//ternary condition to switch player turn
+            }
+
+
         }
 
         public static bool CheckForWin()
         {
-            // your code goes here
+            // declare boolean variable for each win and make a condition of any winning to print the player won and execute DrawBoard method to enter the last mark
+            bool HW = HorizontalWin();
+            bool VW = VerticalWin();
+            bool DW = DiagonalWin();
+            if (HW || VW || DW)
+            {
+                Console.WriteLine("Player " + playerTurn + " won!");
+
+                return true;
+            }
+            else { return false; }
 
-            return false;
-        }
 
+        }
         public static bool CheckForTie()
         {
-            // your code goes here
+            // Since winning is already declared, any other filling of all 9 cells without win is tie. Boolean method should be returned true to make any execution
```

Comment:
This is not really checking for a tie.

Reply:
I see. I will run it and check it. Thanks.